api/ghactions/notify.go

```diff
@@ -175,15 +175,14 @@ func chooseLabels( // nolint:ireturn // TODO: Fix ireturn lint error
 ) mapset.Set {
 	labels := mapset.NewSet()
 
-	if (*workflowRun.Event == "push" &&
-		*workflowRun.HeadRepository.Owner.Login == owner &&
+	// We don't actually check the event here, provided it meets
+	// the criteria to be a run on master.
+	if (*workflowRun.HeadRepository.Owner.Login == owner &&
 		*workflowRun.HeadRepository.Name == repo) &&
 		(*workflowRun.HeadBranch == "master" ||
 			epochBranchesRegex.MatchString(*workflowRun.HeadBranch)) {
 		labels.Add(shared.MasterLabel)
-	}
-
-	if *workflowRun.Event == "pull_request" {
+	} else if *workflowRun.Event == "pull_request" {
```

Comment:
Looking at the JSON for a pull request workflow run targeting the master branch (example: https://github.com/web-platform-tests/wpt/pull/48621)

```sh
$ gh api   -H "Accept: application/vnd.github+json"   -H "X-GitHub-Api-Version: 2022-11-28"   /repos/web-platform-tests/wpt/actions/runs/11379225557
```

```json
{
  "id": 11379225557,
  "name": "manifest",
  "node_id": "WFR_kwLOADc1Vc8AAAACpkEz1Q",
  "head_branch": "orkon/fix-expected-errors",
  "head_sha": "a3e153bfbb2988a03d23a0220b0ed2fb2670787d",
  "path": ".github/workflows/manifest.yml",
  "display_title": "Fix WebDriverAsyncCallbackHandler not expecting all WebDriver exceptions",
  "run_number": 31882,
  "event": "pull_request",
  "status": "completed",
  "conclusion": "success",
  "workflow_id": 3619,
  "check_suite_id": 29729515196,
  "check_suite_node_id": "CS_kwDOADc1Vc8AAAAG7ARmvA",
  "url": "https://api.github.com/repos/web-platform-tests/wpt/actions/runs/11379225557",
  "html_url": "https://github.com/web-platform-tests/wpt/actions/runs/11379225557",
  "pull_requests": [
    {
      "url": "https://api.github.com/repos/web-platform-tests/wpt/pulls/48621",
      "id": 2125373879,
      "number": 48621,
      "head": {
        "ref": "orkon/fix-expected-errors",
        "sha": "a3e153bfbb2988a03d23a0220b0ed2fb2670787d",
        "repo": {
          "id": 3618133,
          "url": "https://api.github.com/repos/web-platform-tests/wpt",
          "name": "wpt"
        }
      },
      "base": {
        "ref": "master",
        "sha": "9cf44ac9a7e1abdaf4aea10cd9d55d13d9e463c1",
        "repo": {
          "id": 3618133,
          "url": "https://api.github.com/repos/web-platform-tests/wpt",
          "name": "wpt"
        }
      }
    }
  ],
  "created_at": "2024-10-17T05:59:44Z",
  "updated_at": "2024-10-17T06:02:21Z",
  "actor": {
    "login": "OrKoN",
    "id": 399150,
    "node_id": "MDQ6VXNlcjM5OTE1MA==",
    "avatar_url": "https://avatars.githubusercontent.com/u/399150?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/OrKoN",
    "html_url": "https://github.com/OrKoN",
    "followers_url": "https://api.github.com/users/OrKoN/followers",
    "following_url": "https://api.github.com/users/OrKoN/following{/other_user}",
    "gists_url": "https://api.github.com/users/OrKoN/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/OrKoN/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/OrKoN/subscriptions",
    "organizations_url": "https://api.github.com/users/OrKoN/orgs",
    "repos_url": "https://api.github.com/users/OrKoN/repos",
    "events_url": "https://api.github.com/users/OrKoN/events{/privacy}",
    "received_events_url": "https://api.github.com/users/OrKoN/received_events",
    "type": "User",
    "user_view_type": "public",
    "site_admin": false
  },
  "run_attempt": 1,
  "referenced_workflows": [],
  "run_started_at": "2024-10-17T05:59:44Z",
  "triggering_actor": {
    "login": "OrKoN",
    "id": 399150,
    "node_id": "MDQ6VXNlcjM5OTE1MA==",
    "avatar_url": "https://avatars.githubusercontent.com/u/399150?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/OrKoN",
    "html_url": "https://github.com/OrKoN",
    "followers_url": "https://api.github.com/users/OrKoN/followers",
    "following_url": "https://api.github.com/users/OrKoN/following{/other_user}",
    "gists_url": "https://api.github.com/users/OrKoN/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/OrKoN/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/OrKoN/subscriptions",
    "organizations_url": "https://api.github.com/users/OrKoN/orgs",
    "repos_url": "https://api.github.com/users/OrKoN/repos",
    "events_url": "https://api.github.com/users/OrKoN/events{/privacy}",
    "received_events_url": "https://api.github.com/users/OrKoN/received_events",
    "type": "User",
    "user_view_type": "public",
    "site_admin": false
  },
  "jobs_url": "https://api.github.com/repos/web-platform-tests/wpt/actions/runs/11379225557/jobs",
  "logs_url": "https://api.github.com/repos/web-platform-tests/wpt/actions/runs/11379225557/logs",
  "check_suite_url": "https://api.github.com/repos/web-platform-tests/wpt/check-suites/29729515196",
  "artifacts_url": "https://api.github.com/repos/web-platform-tests/wpt/actions/runs/11379225557/artifacts",
  "cancel_url": "https://api.github.com/repos/web-platform-tests/wpt/actions/runs/11379225557/cancel",
  "rerun_url": "https://api.github.com/repos/web-platform-tests/wpt/actions/runs/11379225557/rerun",
  "previous_attempt_url": null,
  "workflow_url": "https://api.github.com/repos/web-platform-tests/wpt/actions/workflows/3619",
  "head_commit": {
    "id": "a3e153bfbb2988a03d23a0220b0ed2fb2670787d",
    "tree_id": "cdacd809f4e466ec6d0262f6ee8d5cfb8a23c047",
    "message": "Fix WebDriverAsyncCallbackHandler not expecting all WebDriver exceptions",
    "timestamp": "2024-10-15T14:01:05Z",
    "author": {
      "name": "Alex Rudenko",
      "email": "alexrudenko@chromium.org"
    },
    "committer": {
      "name": "Alex Rudenko",
      "email": "alexrudenko@chromium.org"
    }
  },
  "repository": {
    "id": 3618133,
    "node_id": "MDEwOlJlcG9zaXRvcnkzNjE4MTMz",
    "name": "wpt",
    "full_name": "web-platform-tests/wpt",
    "private": false,
    "owner": {
      "login": "web-platform-tests",
      "id": 37226233,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjM3MjI2MjMz",
      "avatar_url": "https://avatars.githubusercontent.com/u/37226233?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/web-platform-tests",
      "html_url": "https://github.com/web-platform-tests",
      "followers_url": "https://api.github.com/users/web-platform-tests/followers",
      "following_url": "https://api.github.com/users/web-platform-tests/following{/other_user}",
      "gists_url": "https://api.github.com/users/web-platform-tests/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/web-platform-tests/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/web-platform-tests/subscriptions",
      "organizations_url": "https://api.github.com/users/web-platform-tests/orgs",
      "repos_url": "https://api.github.com/users/web-platform-tests/repos",
      "events_url": "https://api.github.com/users/web-platform-tests/events{/privacy}",
      "received_events_url": "https://api.github.com/users/web-platform-tests/received_events",
      "type": "Organization",
      "user_view_type": "public",
      "site_admin": false
    },
    "html_url": "https://github.com/web-platform-tests/wpt",
    "description": "Test suites for Web platform specs — including WHATWG, W3C, and others",
    "fork": false,
    "url": "https://api.github.com/repos/web-platform-tests/wpt",
    "forks_url": "https://api.github.com/repos/web-platform-tests/wpt/forks",
    "keys_url": "https://api.github.com/repos/web-platform-tests/wpt/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/web-platform-tests/wpt/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/web-platform-tests/wpt/teams",
    "hooks_url": "https://api.github.com/repos/web-platform-tests/wpt/hooks",
    "issue_events_url": "https://api.github.com/repos/web-platform-tests/wpt/issues/events{/number}",
    "events_url": "https://api.github.com/repos/web-platform-tests/wpt/events",
    "assignees_url": "https://api.github.com/repos/web-platform-tests/wpt/assignees{/user}",
    "branches_url": "https://api.github.com/repos/web-platform-tests/wpt/branches{/branch}",
    "tags_url": "https://api.github.com/repos/web-platform-tests/wpt/tags",
    "blobs_url": "https://api.github.com/repos/web-platform-tests/wpt/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/web-platform-tests/wpt/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/web-platform-tests/wpt/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/web-platform-tests/wpt/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/web-platform-tests/wpt/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/web-platform-tests/wpt/languages",
    "stargazers_url": "https://api.github.com/repos/web-platform-tests/wpt/stargazers",
    "contributors_url": "https://api.github.com/repos/web-platform-tests/wpt/contributors",
    "subscribers_url": "https://api.github.com/repos/web-platform-tests/wpt/subscribers",
    "subscription_url": "https://api.github.com/repos/web-platform-tests/wpt/subscription",
    "commits_url": "https://api.github.com/repos/web-platform-tests/wpt/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/web-platform-tests/wpt/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/web-platform-tests/wpt/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/web-platform-tests/wpt/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/web-platform-tests/wpt/contents/{+path}",
    "compare_url": "https://api.github.com/repos/web-platform-tests/wpt/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/web-platform-tests/wpt/merges",
    "archive_url": "https://api.github.com/repos/web-platform-tests/wpt/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/web-platform-tests/wpt/downloads",
    "issues_url": "https://api.github.com/repos/web-platform-tests/wpt/issues{/number}",
    "pulls_url": "https://api.github.com/repos/web-platform-tests/wpt/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/web-platform-tests/wpt/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/web-platform-tests/wpt/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/web-platform-tests/wpt/labels{/name}",
    "releases_url": "https://api.github.com/repos/web-platform-tests/wpt/releases{/id}",
    "deployments_url": "https://api.github.com/repos/web-platform-tests/wpt/deployments"
  },
  "head_repository": {
    "id": 3618133,
    "node_id": "MDEwOlJlcG9zaXRvcnkzNjE4MTMz",
    "name": "wpt",
    "full_name": "web-platform-tests/wpt",
    "private": false,
    "owner": {
      "login": "web-platform-tests",
      "id": 37226233,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjM3MjI2MjMz",
      "avatar_url": "https://avatars.githubusercontent.com/u/37226233?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/web-platform-tests",
      "html_url": "https://github.com/web-platform-tests",
      "followers_url": "https://api.github.com/users/web-platform-tests/followers",
      "following_url": "https://api.github.com/users/web-platform-tests/following{/other_user}",
      "gists_url": "https://api.github.com/users/web-platform-tests/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/web-platform-tests/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/web-platform-tests/subscriptions",
      "organizations_url": "https://api.github.com/users/web-platform-tests/orgs",
      "repos_url": "https://api.github.com/users/web-platform-tests/repos",
      "events_url": "https://api.github.com/users/web-platform-tests/events{/privacy}",
      "received_events_url": "https://api.github.com/users/web-platform-tests/received_events",
      "type": "Organization",
      "user_view_type": "public",
      "site_admin": false
    },
    "html_url": "https://github.com/web-platform-tests/wpt",
    "description": "Test suites for Web platform specs — including WHATWG, W3C, and others",
    "fork": false,
    "url": "https://api.github.com/repos/web-platform-tests/wpt",
    "forks_url": "https://api.github.com/repos/web-platform-tests/wpt/forks",
    "keys_url": "https://api.github.com/repos/web-platform-tests/wpt/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/web-platform-tests/wpt/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/web-platform-tests/wpt/teams",
    "hooks_url": "https://api.github.com/repos/web-platform-tests/wpt/hooks",
    "issue_events_url": "https://api.github.com/repos/web-platform-tests/wpt/issues/events{/number}",
    "events_url": "https://api.github.com/repos/web-platform-tests/wpt/events",
    "assignees_url": "https://api.github.com/repos/web-platform-tests/wpt/assignees{/user}",
    "branches_url": "https://api.github.com/repos/web-platform-tests/wpt/branches{/branch}",
    "tags_url": "https://api.github.com/repos/web-platform-tests/wpt/tags",
    "blobs_url": "https://api.github.com/repos/web-platform-tests/wpt/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/web-platform-tests/wpt/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/web-platform-tests/wpt/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/web-platform-tests/wpt/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/web-platform-tests/wpt/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/web-platform-tests/wpt/languages",
    "stargazers_url": "https://api.github.com/repos/web-platform-tests/wpt/stargazers",
    "contributors_url": "https://api.github.com/repos/web-platform-tests/wpt/contributors",
    "subscribers_url": "https://api.github.com/repos/web-platform-tests/wpt/subscribers",
    "subscription_url": "https://api.github.com/repos/web-platform-tests/wpt/subscription",
    "commits_url": "https://api.github.com/repos/web-platform-tests/wpt/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/web-platform-tests/wpt/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/web-platform-tests/wpt/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/web-platform-tests/wpt/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/web-platform-tests/wpt/contents/{+path}",
    "compare_url": "https://api.github.com/repos/web-platform-tests/wpt/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/web-platform-tests/wpt/merges",
    "archive_url": "https://api.github.com/repos/web-platform-tests/wpt/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/web-platform-tests/wpt/downloads",
    "issues_url": "https://api.github.com/repos/web-platform-tests/wpt/issues{/number}",
    "pulls_url": "https://api.github.com/repos/web-platform-tests/wpt/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/web-platform-tests/wpt/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/web-platform-tests/wpt/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/web-platform-tests/wpt/labels{/name}",
    "releases_url": "https://api.github.com/repos/web-platform-tests/wpt/releases{/id}",
    "deployments_url": "https://api.github.com/repos/web-platform-tests/wpt/deployments"
  }
}
```
Head repo name is wpt
Head repo owner is web-platform-tests
branch is orkon/fix-expected-errors 

Verified that `pull_request` events won't match the first conditional (since the first conditional no longer filters by event). This is because on a pull_request event the head branch 

Looks good